Cargo.toml

```diff
@@ -15,6 +15,7 @@ exclude = ["/.github", "/examples", "/scripts"]
 [dependencies]
 tokio = "1.0"
 rustls = { version = "0.23.5", default-features = false, features = ["std"] }
+pin-project-lite = "0.2.14"
```

Comment:
Adding a new dependency is pretty heavy-weight. Can you add some context about why this is required?

Reply:
I just needed it for the test, moved it into dev-dependencies

Reply:
(As dependencies go, pin-project-lite is about as light as they come though.)